backend_kvm.go

```diff
@@ -77,61 +76,38 @@ func (b kvmBackend) KernelRelease() (string, error) {
 	return "", fmt.Errorf("No kernel found")
 }
 
-func (b kvmBackend) hostKernelPath(kernelRelease string) (string, error) {
-	kernelDir := "/boot"
-	kernelPrefix := "vmlinuz-"
-
-	/* First, try to find a kernel with a well-known name */
-	kernelPath := filepath.Join(kernelDir, kernelPrefix + kernelRelease)
-	if _, err := os.Stat(kernelPath); err == nil {
-		return kernelPath, nil
-	}
-
-	/* Otherwise, inspect each kernel installed, and look for the release
-	 * string straight in the binary. Not pretty, but it works. */
-	needle := kernelRelease
-	if !strings.HasSuffix(needle, " ") {
-		// Add space to match exact kernel description string e.g
-		// 4.19.0-6-amd64 (debian-kernel@lists.debian.org) #1 SMP Debian 4.19.67-2+deb10u2 (2019-11-11)
-		needle += " "
-	}
-
-	files, err := ioutil.ReadDir(kernelDir)
+func (b kvmBackend) KernelPath() (string, error) {
+	kernelRelease, err := b.KernelRelease()
 	if err != nil {
 		return "", err
 	}
 
-	for _, f := range files {
-		if !strings.HasPrefix(f.Name(), kernelPrefix) || f.IsDir() {
-			continue
-		}
-
-		kernelPath := filepath.Join(kernelDir, f.Name())
-		buf, err := ioutil.ReadFile(kernelPath)
-		if err != nil {
-			fmt.Fprintln(os.Stderr, "Failed to read kernel:", err)
-			continue
-		}
-
-		if !bytes.Contains(buf, []byte(needle)) {
-			continue
+	/* First we look within the modules directory, as supported by
+	 * various distributions - Arch, Debian, Fedora...
+	 *
+	 * ... perhaps because systemd requires it to allow hibernation
+	 * https://github.com/systemd/systemd/commit/edda44605f06a41fb86b7ab8128dcf99161d2344
+	 */
+	if moddir, err := b.ModulePath(); err == nil {
+		kernelPath := path.Join(moddir, kernelRelease, "vmlinuz")
```

Comment:
Probably `kernelRelease` is redundant here? `moddir` already include kernel release subdir in its path.

Reply:
D'oh sorry about that - fixed with v2.

Note to self: don't submit 3 MRs at the same time - you're bound to miss something.